pkg/controller/appsync/appsync.go

```diff
@@ -43,9 +46,93 @@ func New(config *config.AppSyncConfig, db *database.Database, h *render.Renderer
 }
 
 // HandleSync performs the logic to sync mobile apps.
-func (c *Controller) HandleSync(ctx context.Context) http.Handler {
+func (c *Controller) HandleSync() http.Handler {
+	type AppSyncResult struct {
+		OK     bool    `json:"ok"`
+		Errors []error `json:"errors,omitempty"`
+	}
+
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		// TODO(whaught): implement this
-		controller.InternalError(w, r, c.h, errors.New("not implemented"))
+		logger := logging.FromContext(r.Context()).Named("appsync.HandleSync")
+
+		client := http.Client{Timeout: c.config.Timeout}
+		resp, err := client.Get(c.config.AppSyncURL)
+		if err != nil {
+			controller.InternalError(w, r, c.h, err)
+			return
+		}
+
+		var apps AppsResponse
+		defer resp.Body.Close()
+
+		if err := json.NewDecoder(io.LimitReader(resp.Body, c.config.FileSizeLimitBytes)).Decode(&apps); err != nil {
+			controller.InternalError(w, r, c.h, err)
+			return
+		}
+
+		var merr *multierror.Error
+		realms := map[string]*database.Realm{}
+		appsByRealm := map[uint][]*database.MobileApp{}
+
+		for _, app := range apps.Apps {
+			realm, has := realms[app.Region]
+			if !has {
+				realm, err = c.db.FindRealmByRegion(app.Region)
+				if err != nil {
+					merr = multierror.Append(merr, fmt.Errorf("unable to lookup realm %s: %w", app.Region, err))
+					continue
+				}
+				realms[app.Region] = realm
+			}
+
+			realmApps, has := appsByRealm[realm.ID]
+			if !has {
+				// Only Android supported for sync.
+				realmApps, err := c.db.ListActiveAppsByOS(realm.ID, database.OSTypeAndroid)
```

Comment:
A good follow-up here would be to refactor ListActiveAppsByOS into `ListActiveApps` and a scope 😄